packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/presentation/builders/ExpoNotificationBuilder.java

```diff
@@ -0,0 +1,219 @@
+      builder.setVibrate(NO_VIBRATE_PATTERN);
+    }
+
+    long[] vibrationPatternOverride = getVibrationPatternOverride();
+    if (vibrationPatternOverride != null) {
+      builder.setVibrate(vibrationPatternOverride);
+    }
+
+    if (shouldSetBadge()) {
+      // TODO: Set badge as an effect of presenting notification,
+      //       not as an effect of building a notification.
+      Bundle extras = builder.getExtras();
+      extras.putInt(EXTRAS_BADGE_KEY, getBadgeCount());
+      builder.setExtras(extras);
+    }
+
+    // Add body - JSON data - to extras
+    Bundle extras = builder.getExtras();
+    extras.putString(EXTRAS_BODY_KEY, mNotificationRequest.optString(BODY_KEY));
+    builder.setExtras(extras);
+
+    if (!mNotificationRequest.isNull(THUMBNAIL_URI_KEY)) {
+      try {
+        String thumbnailUri = mNotificationRequest.optString("thumbnailUri");
+        builder.setLargeIcon(mImageLoader.loadImageForDisplayFromURL(thumbnailUri).get(5, TimeUnit.SECONDS));
+      } catch (ExecutionException | InterruptedException e) {
+        Log.w("expo-notifications", "An exception was thrown in process of fetching a large icon.");
+      } catch (TimeoutException e) {
+        Log.w("expo-notifications", "Fetching large icon timed out. Consider using a smaller bitmap.");
+      }
+    }
+
+    return builder;
+  }
+
+  @Override
+  public Notification build() {
+    return createBuilder().build();
+  }
+
+  /**
+   * @return A {@link NotificationChannel}'s identifier to use for the notification.
+   */
+  protected String getChannelId() {
+    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
+      // Returning null on incompatible platforms won't be an error.
+      return null;
+    }
+
+    // We need a channel ID, but we don't know any. Let's use system-provided one as a fallback.
+    Log.w("ExpoNotificationBuilder", "Using `NotificationChannel.DEFAULT_CHANNEL_ID` as channel ID for push notification. " +
+        "Please provide a NotificationChannelsManager to provide builder with a fallback channel ID.");
+    return NotificationChannel.DEFAULT_CHANNEL_ID;
+  }
+
+  private int getBadgeCount() {
+    return mNotificationRequest.optInt(BADGE_KEY);
+  }
+
+  private boolean shouldShowAlert() {
+    return !mNotificationRequest.isNull(CONTENT_TITLE_KEY) || !mNotificationRequest.isNull(CONTENT_TEXT_KEY);
+  }
+
+  private boolean shouldPlaySound() {
+    return mNotificationRequest.optBoolean(SOUND_KEY);
+  }
+
+  private boolean shouldUseDefaultVibration() {
+    return mNotificationRequest.optBoolean(SOUND_KEY) || mNotificationRequest.optBoolean(VIBRATE_KEY);
+  }
+
+  private boolean shouldSetBadge() {
+    return !mNotificationRequest.isNull(BADGE_KEY);
+  }
+
+  private Number getPriorityOverride() {
+    if (mNotificationRequest.isNull(PRIORITY_KEY)) {
+      return null;
+    }
+
+    switch (mNotificationRequest.optString(PRIORITY_KEY)) {
```

Comment:
sorry to be pedantic, but could we also allow setting `PRIORITY_DEFAULT` explicitly? 😅

Reply:
Sorry, following your suggestion that _we allow this in the current API_ I just copied this `switch` from `expoview` not verifying if it covers all values. Fixing!